File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/section_3_3_2_ocp/after/Play.java
package com.grokkingsoftwarearchitecture.chapter03.section_3_3_2_ocp.after;

/**
 * The Architectural Boundary (The Contract).
 * * ARCHITECTURE NOTE: This interface is the secret to the Open/Closed Principle. 
 * By defining a standard contract that all plays must follow, we create a "seam" 
 * in our architecture. The Midfielder doesn't need to know the details of the play, 
 * it only needs to know that the play has an execute() method.
 */
public interface Play {
    void execute();
}

File: Chapter 03/Python/section_3_3_2_ocp/after/play.py
from abc import ABC, abstractmethod

class Play(ABC):
    """
    The Architectural Boundary (The Contract).

    ARCHITECTURE NOTE: This interface is the secret to the Open/Closed Principle.
    By defining a standard contract that all plays must follow, we create a "seam"
    in our architecture. The Midfielder doesn't need to know the details of the play,
    it only needs to know that the play has an Execute() method.
    """
    @abstractmethod
    def execute(self):
        pass
